include/onnxruntime/core/optimizer/graph_transformer.h

```diff
@@ -117,7 +127,28 @@ class RuleBasedGraphTransformer : public GraphTransformer {
     return nullptr;
   }
 
+  /*
+  Get rewrite rules that get applied to a node regardless of its op_type ("any-op" rewrite rules).
+  @returns a pointer to the vector containing all any-op rewrite rules, nullptr otherwise.
+  */
+  const std::vector<std::unique_ptr<RewriteRule>>* GetAnyOpRewriteRules() const {
+    return GetRewriteRules(kAnyOpRewriteRules);
+  }
+
+ protected:
+  /** Apply the given set of rewrite rules on the Node of this Graph.
+  @param[in] graph The Graph.
+  @param[in] node The Node to apply the rules to.
+  @param[in] rules The vector of RewriteRules that will be applied to the Node.
+  @param[out] modified Set to indicate whether the node was modified or not.
+  @param[out] deleted Set to indicate if the node was deleted.
+  @returns Status indicating success or providing error information */
+  common::Status ApplyRulesOnNode(Graph& graph, Node& node,
+                                  const std::vector<std::unique_ptr<RewriteRule>>& rules,
+                                  bool& modified, bool& deleted) const;
+
  private:
+  static constexpr const char* kAnyOpRewriteRules = "AnyOpRewriteRules";
   using RewriteRuleSet = std::unordered_map<std::string, std::vector<std::unique_ptr<RewriteRule>>>;
```

Comment:
Agree. with a set of rules, it is easy to construct such a map if needed.

Reply:
By having a list for all rules, there will be no trigger for calling a rule. For each node of the graph, we will have go over all rules and check if their condition is satisfied. With the current hashmap approach, we do a very quick pre-filtering based on the op type of the node and then check for further conditions.
There are some rules (such as constant folding) that should be triggered for every op type, but this seems okay to me (I would not penalize all other rules for that). I could also turn the constant folding to a graph transformer on its own, but there might be another rule later that has to be called for all op types.

Reply:
I addressed this as per the last commit. There is now a single vector that holds all rules registered in a specific rule-based transformer (although I still preferred the quick filtering of the hashmap :)). I tried to organize things a bit by creating a method that for each rule checks op type compatibility. I have some ideas to make this more structured, but let's make it more complicated when it is needed.

Reply:
I strongly agree with @kkaranasos. Such a change will affect all re-write rules... Constant Folding seems to be an exception and should be written as a transformer instead if it is applicable to all the nodes. By moving away from a map now for every node we are checking every single rule which is registered and this is unnecessary.

Reply:
The map between node type and rule is based on the assumption that rewrite rule will be defined based on node type. But, this assumption is not valid. The graph optimization is supposed to be done based on some patterns(node type, constant inputs, ...), which will be defined as rules. And as Yuan said, if it is really needed, this kind of map can be constructed later. But now, the code should not be made too complicate for an uncertain assumption. 

Indeed, implementing constant folding as rewrite rule is right way since it is local transformation. In the original code review feedback, I was suggesting to change to implement it as transformer because the constant folding was implemented by creating a session for each node.

Reply:
Let's get this simpler version in. It also improves the API, since as we discussed we should not need to pass the op type when registering a rule. Once we have more rules, let's revisit what is the most efficient way to trigger them.